lib/functions.php

```diff
@@ -37,6 +37,9 @@ function digest_site(ElggUser $user, $interval) {
 	
 	// store current user
 	$current_user = elgg_get_logged_in_user_entity();
+	if ($current_user == null) {
+		$is_impersonate_session = login($user);
```

Comment:
we can't use the function `login()` as there are events fired when you call this function. we don't want that to happen